### PR TITLE
workflow execution summary

### PR DESCRIPTION
**Pull request by danielnorberg**
# Hey, I just made a Pull Request!

## Description
Make `FloRunner.runTask()` log this after executing:

```
Executed 3 out of 4 tasks:
bar(2018-09-14T00:00:00Z,2018-09-13T00:00:00Z)#d420acb8: Pending
├▸ foo(2018-09-13T00:00:00Z)#fdd54b4d: Success
│  └▸ baz(2018-09-14T00:00:00Z)#fc6d8b0e: Success
└▸ quux(2018-09-14T00:00:00Z)#8250510a: Failure: java.lang.RuntimeException: Failure!
```

## Motivation and Context
Give users a summary of the entire workflow execution.

## Checklist for PR author(s)

- [ ] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

**Comment by honnix** [comment]
surprisingly a simple change.

File: flo-runner/src/main/java/com/spotify/flo/context/Logging.java [hunk 1]
```diff
@@ -101,4 +123,17 @@ void printPlan(TaskInfo taskInfo) {
     PrintUtils.tree(taskInfo).forEach(LOG::info);
     LOG.info("");
   }
+
+  @AutoValue
+  static abstract class Status {
+    abstract boolean success();
+    abstract Optional<Throwable> failure();
```
Comment (ulzha) [inline flo-runner/src/main/java/com/spotify/flo/context/Logging.java]: Could we just have absence of failure mean success? So "Unknown Error" printout is prevented by design.

**Comment by ulzha** [comment]
Looks like they'll love it.

File: flo-runner/src/main/java/com/spotify/flo/context/Logging.java (hunk 1 above)
Comment (danielnorberg) [reply flo-runner/src/main/java/com/spotify/flo/context/Logging.java]: 👍

File: flo-runner/src/main/java/com/spotify/flo/context/Logging.java [hunk 2]
```diff
@@ -78,8 +86,22 @@ void failedValue(TaskId taskId, Throwable valueError, Duration elapsed) {
...
+      final Status status = statuses.get(taskId);
+      final String statusLine;
+      if (status == null) {
+        statusLine = "Pending";
+      } else if (status.success()) {
+        statusLine = "Success";
+      } else {
+        statusLine = "Failure: " + status.failure().map(Throwable::toString).orElse("Unknown Error");
+      }
+      LOG.info("{}: {}", s, statusLine);
```
Comment (danielnorberg) [inline flo-runner/src/main/java/com/spotify/flo/context/Logging.java]: Maybe the summary should be a single multi-line log message?

Comment (honnix) [reply flo-runner/src/main/java/com/spotify/flo/context/Logging.java]: Makes sense. Although it would be multiple lines either way in stackdriver.

Comment (danielnorberg) [reply flo-runner/src/main/java/com/spotify/flo/context/Logging.java]: A single entry with multiple lines might be easer for users to view & copy-paste etc.

**Comment by danielnorberg** [comment]
Maybe good enough to merge as an initial implementation?

**Comment by honnix** [comment]
With multiple log entries?

**Comment by honnix** [comment]
Maybe that's good enough. Multiline might make the log entry look weird.

**Comment by honnix** [comment]
I think this has a good enough state to be shipped. 👍 :shipit:

**Pull request merged**